From 22a9eae39eafd63cee02ff7ef3ffb02336bd2a29 Mon Sep 17 00:00:00 2001
From: nikhilad <nikhilad@ami.com>
Date: Mon, 30 Jun 2025 07:51:52 +0530
Subject: [PATCH] Added Restart_cause for Power Down

Added Restart casue as Watchdog Timer
expired for Power Down

Signed-off-by: nikhilad <nikhilad@ami.com>
---
 src/watchdog.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/watchdog.cpp b/src/watchdog.cpp
index ae9f4e3..f921aaa 100644
--- a/src/watchdog.cpp
+++ b/src/watchdog.cpp
@@ -332,7 +332,8 @@ void Watchdog::timeOutHandler()
             }
             else
             {
-                if (action == Watchdog::Action::PowerCycle)
+                if ((action == Watchdog::Action::PowerCycle) ||
+                    (action == Watchdog::Action::PowerOff))
                 {
                     auto method = bus.new_method_call(
                         restart::busName, restart::path,
-- 
2.25.1

